x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -232,19 +238,134 @@ private static void validateField(Map<?, ?> properties, String fieldName, boolea
+            resolveRangeTypeEnrichMapping(policy, resultListener);
         } else if (EnrichPolicy.GEO_MATCH_TYPE.equals(policy.getType())) {
-            matchFieldMapping = (builder) -> builder.field("type", "geo_shape");
+            createEnrichMappingBuilder((builder) -> builder.field("type", "geo_shape"), resultListener);
         } else {
-            throw new ElasticsearchException("Unrecognized enrich policy type [{}]", policy.getType());
+            resultListener.onFailure(new ElasticsearchException("Unrecognized enrich policy type [{}]", policy.getType()));
         }
+    }
+
+    private void resolveRangeTypeEnrichMapping(EnrichPolicy policy, ActionListener<XContentBuilder> resultListener) {
+        GetFieldMappingsRequest fieldMappingsRequest = new GetFieldMappingsRequest().indices(
+            policy.getIndices().stream().toArray(String[]::new)
+        ).fields(policy.getMatchField());
+
+        client.execute(GetFieldMappingsAction.INSTANCE, fieldMappingsRequest, listener.delegateFailure((l, response) -> {
+            String matchField = policy.getMatchField();
+            List<String> types = collapseFieldMappingsValue(response, toFieldMappingType(matchField));
+            if (types.isEmpty()) {
+                resultListener.onFailure(
+                    new ElasticsearchException(
+                        "No mapping type found for match field '{}' - indices({})",
+                        policy.getMatchField(),
+                        Strings.collectionToCommaDelimitedString(policy.getIndices())
+                    )
+                );
+            } else if (types.size() > 1) {
+                resultListener.onFailure(
+                    new ElasticsearchException(
+                        "Multiple distinct mapping types for match field '{}' - indices({})  types({})",
+                        policy.getMatchField(),
+                        Strings.collectionToCommaDelimitedString(policy.getIndices()),
+                        Strings.collectionToCommaDelimitedString(types)
+                    )
+                );
+            } else {
+                String matchType = resolveMatchType(types.get(0));
+                if (matchType == null) {
+                    resultListener.onFailure(
+                        new ElasticsearchException(
+                            "Field '{}' has type [{}] which doesn't appear to be a range type",
+                            policy.getMatchField(),
+                            types.get(0)
+                        )
+                    );
+                } else {
+                    // date_range types mappings allow for the format to be specified, should be preserved in the created index
+                    if (matchType.equals("date_range")) {
+                        List<String> formatEntries = collapseFieldMappingsValue(response, toFielMappingFormat(matchField));
+                        if (types.isEmpty()) {
```

Comment:
Shouldn't this check ``formatEntries.isEmpty()` instead?

Reply:
Good catch, addressed in https://github.com/elastic/elasticsearch/pull/76110/commits/910db97ce4ddf0dace3c81415d9ceabc42bd3062